libraries/node-core-library/src/FileSystem.ts

```diff
@@ -0,0 +1,432 @@
+import * as pathUtilities from 'path';
+import * as fs from 'fs';
+import * as fsx from 'fs-extra';
+
+import { Text } from './';
+
+/**
+ * The allowed types of encodings, as supported by Node.js
+ * @public
+ */
+export type Encoding = 'utf8';
+
+/**
+ * Available file handle opening flags.
+ * @public
+ */
+export type FileHandleOpenFlags = 'r' | 'r+' | 'rs+' | 'w' | 'wx' | 'w+' | 'wx+' | 'a' | 'ax' | 'a+' | 'ax+';
+
+/**
+ * Enumeration controlling conversion of newline characters.
+ * @public
+ */
+export enum NewlineConversion {
```

Comment:
>enum [](start = 7, length = 4)

Should this be a const enum, maybe with the strings themselves as the values (?)